hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionCoprocessorHost.java

```diff
@@ -409,6 +432,18 @@ void loadTableCoprocessors(final Configuration conf) {
   @Override
   public RegionEnvironment createEnvironment(RegionCoprocessor instance, int priority, int seq,
       Configuration conf) {
+    if (coprocessorService instanceof HCompactionServer) {
```

Comment:
Could we add some configs to avoid instanceof here, or we sub class RegionCoprocessorHost?

Reply:
Yes, We can introduce `RegionCompactionCoprocessorHost` which only used for CompactionServer